app/models/balance.rb

```diff
@@ -56,4 +56,15 @@ def payments_as_beneficiary
   def payments_as_payer
     @payments_as_payer ||= Payment.newest_first.where(payer: user)
   end
+
+  def transfers_as_payer
+    @transfers_as_payer ||= Transfer.where(from_id: user.id, status: :pending)
+  end
+
+  def creditors
+    payments_as_beneficiary
+      .map(&:payer)
+      .uniq
+      .select { |user| balance_for(user) < 0 }
```

Comment:
This looks like a big N+1. The `balance_for` makes 3 requests to the database for each creditor. I'm not very familiar with how exactly are the models built but to me it looks like either a case for a custom SQL or at least a cache that holds these values and is invalidated each time a payment occurs that involves the current user.